mongo/change_stream.go

```diff
@@ -457,20 +457,33 @@ func (cs *ChangeStream) ResumeToken() bson.Raw {
 }
 
 // Next gets the next result from this change stream. Returns true if there were no errors and the next
-// result is available for decoding.
+// result is available for decoding. Next blocks until an event is available for decoding or ctx expires.
```

Comment:
Maybe talk about what happens if ctx expires -- you get an error and what happens to the ChangeStream?  Is it valid?  Will it have to restart?

Reply:
I'm not 100% sure. My initial thought is that we can't guarantee the cursor is in a valid state because we could have a situation where we send a `getMore` request but the context expires before we get the response. In this case, the server has already moved onto the next batch and we'll skip a batch of events. Thoughts?